tools/configure_server_settings.rb

```diff
@@ -69,8 +69,11 @@
 if opts[:force]
   puts "Change [#{opts[:path]}], old class: [#{path[key].class}], new class: [#{newval.class}]"
 elsif path[key] && path[key].class != newval.class
-  STDERR.puts "The new value's class #{newval.class} does not match the prior one's #{path[key].class}. Use -t to specify the type for the provided value. Use -f to force changing this value. Note, -f may break things! See -h for examples."
-  exit 1
+  # If both types are boolean, it is ok, if they differ
+  unless [path[key], newval].all? { |e| [TrueClass, FalseClass].include?(e.class) }
```

Comment:
This might be clearer if the comparison is moved to a method..otherwise the elsif combined with an inner unless is just confusing...how about...

```ruby
def types_match?(last_val, new_val)
  case last_val
  when TrueClass, FalseClass
    new_val.is_a?(TrueClass) || new_val.is_a?(FalseClass)      # Alternately, a more clever way: !!new_val == new_val
  else
    !new_val.is_a?(last_val.class)
  end
end

...

if opts[:force]
  ...
elsif path[key] && !types_match(path[key], newval)
  STDERR.puts ...
  exit 1
end
```

Reply:
Yes, that's way better. (Apparently my box was really small, since the possibility of using a method did not occur to me 😃 )

However to me the case-statement is still a bit not trivial. Since there is apparently no short way to check if something is a boolean, I put that piece of code in it's own method too